rotkehlchen/tasks/manager.py

```diff
@@ -696,6 +700,29 @@ def _maybe_update_morpho_cache(self) -> Optional[list[gevent.Greenlet]]:
 
         return greenlets if len(greenlets) > 0 else None
 
+    def _maybe_update_pendle_cache(self) -> Optional[list[gevent.Greenlet]]:
+        with self.database.conn.read_ctx() as cursor:
+            if (
```

Comment:
Hmmm this seems wrong. Why do 5 queries in the DB? You only use `get_single_blockchain_addresses` if you want to specifically access only a single chain.

What you need is `get_blockchain_accounts` which returns all the data in one struct with one call, right? Then you can just read whatever you want. Or even you can make a custom DB call for this if you want. But keep it in 1 query. So some function called `has_tracked_addresses_in_chains(chains: list[SupportedBlockchain])` or something like that

Then also replace it in other places it's used wrongly. I see you probably just copied pasted the logic from other places in this file. Fix it in all other places such as `_maybe_update_aura_pools` etc.